### DISCUSSION TITLE
opstrace within k8s cluster

### DISCUSSION THREAD

**Discussion by devopstoday11** [opstrace/opstrace#529] [Ideas]
Hi,
Really useful product. Nice job!
I see your docs have only steps for installing opstrace on a VM. Are you guys not supporting k8s deployment of opstrace yet (opstrace helm chart or opstrace operator)?  If opstrace can be setup on k8s cluster instead of a regular VM environment, it would be easy to use opstrace in any cloud provider and also on-premise. 
Do you have a roadmap to get opstrace deployed on k8s? 

Thank you
Chandu

**Comment by triclambert** [comment]
Hi @devopstoday11, I replied to your direct email several days ago.  Just seeing now that you also posted this discussion message.

Today we do provision Kubernetes clusters under the hood (using EKS or GKE).  We consider that an implementation detail, though.  Our reasoning is that observability should be run separately from the workloads to be observed.  It is also easier for us to test and validate correctness in a stable environment that we can re-create identically, end-to-end, each time tests run.  Basically, you just run `opstrace create` and we take care of the rest.

So while at this time we do not have installing Opstrace on existing Kubernetes clusters on our roadmap, we are considering adding it.  Can you share a bit about your use case?
